### PR TITLE
Use subtest_streamed for the sequential executor [WIP]

### PR DESCRIPTION
**Pull request by oschwald**
We have been suffering with #100 since upgrading to a modern version of `Test::Class::Moose`. We currently use a hacked up formatter to avoid the issue, but that is not ideal.

This change makes the sequential executor use `subtest_streamed` instead of `subtest_streamed`. In my testing, this works as expected.

The tests have not been updated yet as it will be fairly extensive to change all of the `test_events_is` checks. I wanted to make sure the general approach was acceptable first. Also, if there is an easy way to dump the events for those checks, it would likely save quite a bit of manual labor.

### CODE REVIEW

**Comment by autarch** [comment]
This seems fine to me. This doesn't seem like it would break anyone's use of this package.

I just pushed a CI fix to master, so you'll need to rebase off master to get the tests to run.

As far as dumping the events, I think I used https://metacpan.org/pod/Test2::Tools::EventDumper when I converted TCM to Test2. The author has marked it as deprecated but I think it'll still work for your purposes. I also wrote a `dump_events` sub in `t/lib/Test/Events.pm` that might be helpful.

From a quick look, I don't think the changes will be _too_ terrible. It looks like this just adds an additional Note event at the beginning of every subtest.

**Comment by oschwald** [comment]
I am not sure how to read the two failures, but I think they are unrelated.

**Comment by autarch** [comment]
If you're not rebased off the latest master that might explain the test failure. Otherwise it means you need to run `tidyall` on the code.

**Comment by oschwald** [comment]
It might be nice to get rid of the `Subtest: ` prefix. Looking at how `subtest_streamed` is implemented, we could probably call `Test2::API::run_subtest` ourselves or alternatively call `subtest_buffered` with `buffered => 0`.

**Comment by oschwald** [comment]
There were two `Test::Vars` failures unrelated to my changes that I fixed, but otherwise `tidyall` is clean for me.

**Comment by autarch** [comment]
That `Subtest: ` prefix is coming from Test2 itself, right? I think we'd have to subclass/replace some part of it to remove that. If it's not too gross to do so, that'd be great, but it's not worth doing some sort of internals mucking about that's likely to break.

**Comment by oschwald** [comment]
Yeah, the prefix is coming from `subtest_streamed` itself. I just pushed a commit showing how it might look without it.

**Comment by oschwald** [comment]
I realized that I had failed to push my last commit. I pushed it now. I think this is ready for review.

**Review by autarch — CHANGES_REQUESTED** [review]
Instead of splitting all the test classes' `expected_test_events` into two methods, I think it'd be better to just pass in an argument indicating that the tests are being run in parallel.

AFAICT, all you'd need to do for this flag is something like this:

```perl
    event Note => sub {
        call message => 'TestsFor::Basic';
    } unless $is_parallel;
```

This will make it much simpler to change these tests in the future if that's needed.

**Comment by oschwald** [comment]
My feeling was that the tests are already extremely hard to follow and update, and introducing arguments like that would make the situation even worse. Right now it is somewhat doable to update the tests by dumping the huge set of events, but once you introduce arguments and the logic for the number of cases that the two event trees differ, then that will be impossible.

That said, I think this whole method of testing even outputs is misguided. It makes the tests brittle and hard to update.

**Comment by autarch** [comment]
> My feeling was that the tests are already extremely hard to follow and update, and introducing arguments like that would make the situation even worse. Right now it is somewhat doable to update the tests by dumping the huge set of events, but once you introduce arguments and the logic for the number of cases that the two event trees differ, then that will be impossible.
> 
> That said, I think this whole method of testing even outputs is misguided. It makes the tests brittle and hard to update.

Do you have any ideas on how to make the tests easier to work with? One possibility that comes to mind is to only test the full set of events for a few things, and then just look for a smaller subset in most tests, since we're often just checking a specific portion of the test output in each test.
